### ISSUE TITLE
Facility selector should have an accessible name

### ISSUE THREAD

**Issue by nathancrtr** [CDCgov/prime-simplereport#4020]
## Area or Component
Conduct Test

## Quick Summary
The facility selector doesn't have an accessible name.

## Accessibility Violation
Select element must have an accessible name (https://dequeuniversity.com/rules/axe/4.4/select-name?application=AxeChrome)

## Action Requested
The facility select dropdown in the site header should be named and labeled.

The dropdown does not have a visible label, so instead of a `<label>` tag around the dropdown. Labeling the select element itself with an invisible `aria-label` should do the trick.

## Source Code (optional)
https://github.com/CDCgov/prime-simplereport/blob/34cd6da49e112854b1c4a0832354e816b228fbd7/frontend/src/app/commonComponents/Header.tsx#L363-L372
